examples/generate.sh

```diff
@@ -102,13 +102,23 @@ echo "Generated ${CONTROLPLANE_GENERATED_FILE}"
 kustomize build "${SOURCE_DIR}/machinedeployment" | envsubst >> "${MACHINEDEPLOYMENT_GENERATED_FILE}"
 echo "Generated ${MACHINEDEPLOYMENT_GENERATED_FILE}"
 
-# Generate Cluster API provider components file.
-curl -L https://github.com/kubernetes-sigs/cluster-api/releases/download/v0.2.3/cluster-api-components.yaml > "${COMPONENTS_CLUSTER_API_GENERATED_FILE}"
-echo "Downloaded ${COMPONENTS_CLUSTER_API_GENERATED_FILE}"
+  # Generate Cluster API provider components file.
+if [[ ${CAPI_BRANCH:-""} == "stable" ]]; then
+  curl -L https://github.com/kubernetes-sigs/cluster-api/releases/download/v0.2.3/cluster-api-components.yaml > "${COMPONENTS_CLUSTER_API_GENERATED_FILE}"
+  echo "Downloaded ${COMPONENTS_CLUSTER_API_GENERATED_FILE}"
+else
+  kustomize build "github.com/kubernetes-sigs/cluster-api/config/default/?ref=${CAPI_BRANCH}" > "${COMPONENTS_CLUSTER_API_GENERATED_FILE}"
+  echo "Generated ${COMPONENTS_CLUSTER_API_GENERATED_FILE}"
+fi
 
 # Generate Kubeadm Bootstrap Provider components file.
+if [[ ${CABPK_BRANCH:-""} == "stable" ]]; then
```

Comment:
Do we want the unset case to use the known version of the components? If not, then the reference to CABPK_BRANCH below could fail...

Reply:
got it. added a default now.